mikrotik/resource_interface_wireguard_peer.go

```diff
@@ -0,0 +1,273 @@
+	var plan interfaceWireguardPeerModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updated, err := i.client.UpdateInterfaceWireguardPeer(modelToInterfaceWireguardPeer(&plan))
+	if err != nil {
+		resp.Diagnostics.AddError("update failed", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(interfaceWireguardPeerToModel(updated, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (i *interfaceWireguardPeer) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state interfaceWireguardPeerModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := i.client.DeleteInterfaceWireguardPeer(state.ID.ValueString()); err != nil {
+		resp.Diagnostics.AddError("Could not delete interfaceWireguardPeer", err.Error())
+		return
+	}
+}
+
+func (i *interfaceWireguardPeer) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Retrieve import ID and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("interface"), req, resp)
+}
+
+type interfaceWireguardPeerModel struct {
+	ID                     tftypes.String `tfsdk:"id"`
+	AllowedAddress         tftypes.String `tfsdk:"allowed_address"`
+	Comment                tftypes.String `tfsdk:"comment"`
+	Disabled               tftypes.Bool   `tfsdk:"disabled"`
+	EndpointAddress        tftypes.String `tfsdk:"endpoint_address"`
+	EndpointPort           tftypes.Int64  `tfsdk:"endpoint_port"`
+	Interface              tftypes.String `tfsdk:"interface"`
+	PersistentKeepalive    tftypes.Int64  `tfsdk:"persistent_keepalive"`
+	PresharedKey           tftypes.String `tfsdk:"preshared_key"`
+	PublicKey              tftypes.String `tfsdk:"public_key"`
+	CurrentEndpointAddress tftypes.String `tfsdk:"current_endpoint_address"`
+	CurrentEndpointPort    tftypes.Int64  `tfsdk:"current_endpoint_port"`
+}
+
+func interfaceWireguardPeerToModel(i *client.InterfaceWireguardPeer, m *interfaceWireguardPeerModel) diag.Diagnostics {
+	var diags diag.Diagnostics
+	if i == nil {
+		diags.AddError("Interface Wireguard Peer cannot be nil", "Cannot build model from nil object")
+		return diags
+	}
+	m.ID = tftypes.StringValue(i.Id)
+	m.AllowedAddress = tftypes.StringValue(i.AllowedAddress)
+	m.Comment = tftypes.StringValue(i.Comment)
+	m.Disabled = tftypes.BoolValue(i.Disabled)
+	m.EndpointAddress = tftypes.StringValue(i.EndpointAddress)
+	m.EndpointPort = tftypes.Int64Value(int64(i.EndpointPort))
+	m.Interface = tftypes.StringValue(i.Interface)
+	m.PersistentKeepalive = tftypes.Int64Value(int64(i.PersistentKeepalive))
```

Comment:
Updating `PersistentKeepalive` and `EndpointPort` to use `int64` data types rather than `int` data types should allow you to remove the `int64( .. )` conversion here and the reverse conversion in `modelToInterfaceWireguardPeer`.